### PR TITLE
Cleanup/improve Plist usage and Event API

### PR DESCRIPTION
**Pull request by jsiwek**
`topic/jsiwek/plist-and-event-cleanup` is in `zeek` and `bifcl` repos.   See commit message for details, but the bulk of this is motivated toward getting rid of some of the extra or unnecessary heap allocation being done in most places that generate/queue events and their argument Vals.

Final measurements on a test pcap were giving ~2.5% better timing and ~12% reduction in max mem. allocated (but this was only a small test case that ends up allocating on the order of 150mb total, I'd expect cases where the bulk of memory is used for storing state in script-land, there's less benefit to optimizing how much memory gets used for storing the current-batch-of-queued-events, but still).

### CODE REVIEW

File: src/Conn.h [hunk 1]
```diff
@@ -181,6 +181,8 @@ class Connection : public BroObj {
 				val_list* vl);
 	void ConnectionEvent(EventHandlerPtr f, analyzer::Analyzer* analyzer,
 				val_list vl);
+	void ConnectionEventFast(EventHandlerPtr f, analyzer::Analyzer* analyzer,
```
Comment (0xxon) [inline src/Conn.h]: Would you mind adding documentation to ConnectionEvent & ConnectionEventFast?

While just ConnectionEvent itself was kind of self-documentary, I expect that having ConnectionEvent + ConnectionEventFast will be rather confusing to people not familiar with the background :)

File: src/Event.h [hunk 2]
```diff
@@ -58,6 +58,13 @@ class EventMgr : public BroObj {
 	EventMgr();
 	~EventMgr() override;
 
+	void QueueEventFast(const EventHandlerPtr &h, val_list vl,
```
Comment (0xxon) [inline src/Event.h]: Similar comment here - I think with the existence of QueueEventFast we now need documentation that explains the difference betwen QueueEvent & QueueEventFast exactly.

I know it is basically just an existence check - but without a comment like in the commit-message it is hard to get what is going on here.

**Comment by 0xxon** [comment]
This looks basically good to me - if you could add documentation to 2 of the Fast methods (see above) that would be great.

Also - what exactly happens if someone uses one of the new methods when they shouldn't (so queuing an event where the handler is false)?

**Comment by jsiwek** [comment]
> This looks basically good to me - if you could add documentation to 2 of the Fast methods (see above) that would be great.

Sure.

> Also - what exactly happens if someone uses one of the new methods when they shouldn't (so queuing an event where the handler is false)?

A memory leak.

**Comment by 0xxon** [comment]
Thanks a lot - I will merge this before EOD today.

**Pull request merged**